.github/workflows/ci.yaml

```diff
@@ -0,0 +1,134 @@
+name: build
+
+on:
+  push:
+    branches: [ testing ]
+  pull_request:
+    branches: [ testing ]
+
+jobs:
+  java:
+    runs-on: ubuntu-latest
+    # Steps represent a sequence of tasks that will be executed as part of the job
+    steps:
+    # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+    - name: 'deployment pending'
+      uses: 'chrnorm/deployment-action@releases/v1'
+      id: deployment
+      with:
+        token: '${{ github.token }}'
+        environment: 'production'
+        target_url: 'https://gcp.elsa-hilft.ch'
+    - uses: actions/checkout@v2
+    - uses: actions/cache@v1
+      with:
+        path: ~/.gradle/caches
+        key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*') }}
+        restore-keys: |
+          ${{ runner.os }}-gradle-
+    - uses: actions/cache@v1
+      with:
+        path: backend/build/jib-cache
+        key: ${{ runner.os }}-jib
+        restore-keys: |
+          ${{ runner.os }}-jib
+    - name: Set up JDK 1.11
+      uses: actions/setup-java@v1
+      with:
+        java-version: 1.11
+    - name: Build image
+      run: |
```

Comment:
Build & Push can be a single step using `jib` https://github.com/GoogleContainerTools/jib/tree/master/jib-gradle-plugin#configuration

Reply:
Yes, buuut it requires a credential helper AFAIU(and I'm not completely sure how the GitHub auth works), so I would leave that for later.